### ISSUE TITLE
biquery/datatransfer/apiv1: ListTransferConfigsRequest.Project not respected for service accounts

### ISSUE THREAD

**Issue by carlpett** [googleapis/google-cloud-go#1490]
## Client

DataTransfer

## Describe Your Environment

Any. Tested local dev, local docker, GKE

## Expected Behavior

Calling `datatransferClient.ListTransferConfigs(ctx, &datatransferpb.ListTransferConfigsRequest{Parent: fmt.Sprintf("projects/%s", project)})` should list transfer configurations of `projects/<project>`.

## Actual Behavior

It works if I make the call using my own gsuite credentials. However, if using a service account, the call lists transfer configurations in the service account's "home project" (and in my case, fails, because that project does not have that API enabled).

**Comment by odeke-em** [comment]
Thank you for reporting this issue @carlpett and welcome to the Google-Cloud-Go project!

So I see you are filing against bigquery/datatransfer/apiv1. I am not a bigquery expert but let me ping @jadekler to perhaps help find a bigquery expert and I'll also ask.

**Comment by jadekler** [comment]
cc @shollyman

**Comment by shollyman** [comment]
Thanks for reporting this.  I've filed https://issuetracker.google.com/issues/137010966 on the BigQuery public issue tracker to have the backend team address this. I've confirmed the ListTransferConfigs behavior is incorrect as you've noted when leveraging service account credentials.

As this github repository primarily services issues with the libraries themselves, I'll go ahead and close out this issue in favor of the one on the public issue tracker.
